lib/web/users.go

```diff
@@ -0,0 +1,126 @@
+/*
+Copyright 2020 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package web
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/gravitational/teleport/lib/auth"
+	"github.com/gravitational/teleport/lib/httplib"
+	"github.com/gravitational/teleport/lib/reversetunnel"
+	"github.com/gravitational/teleport/lib/services"
+	"github.com/gravitational/trace"
+
+	"github.com/julienschmidt/httprouter"
+)
+
+// requestUser is used to pull out data from JSON requests.
+type requestUser struct {
```

Comment:
how about `requestCreateUser`?

Reply:
i wanted to make this struct generic for reuse for other requests like `delete` and `update`, both of these requests will also require at least the `name` prop

Reply:
but `delete` won't use `roles`, right?
or maybe `delete` should use HTTP `DELETE` verb on `/webapi/sites/:site/namespaces/:namespace/users/:name`.

anyway, this is ok to leave as is

Reply:
```golang
saveUserRequest {
  isNew,
  name,
  roles,
}

createResetPasswordTokenRequest {
 //
}

func (h *Handler) saveUser(..)
func (h *Handler) createResetPasswordToken(..)

```


And when creating a user on the client side, make  2 separate calls:

```
saveUser(..., isNew)
createResetToken(...)
```

(we are going to add an ability to reset passwords anyway)

This will eliminate the need for this 

```
type responseCreateUser struct {
```